servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpServerBuilder.java

```diff
@@ -93,11 +105,37 @@ public interface HttpServerBuilder {
      * Implementations can round the specified {@link Duration} to full time units, depending on their time granularity.
      * {@link Duration#ZERO Zero (0)} disables timeout.
      * @return {@code this}.
+     * @deprecated use {@link #sslConfig(ServerSslConfig, Map, int, Duration, boolean)} instead.
      */
+    @Deprecated
     default HttpServerBuilder sslConfig(ServerSslConfig defaultConfig, Map<String, ServerSslConfig> sniMap,
                                         int maxClientHelloLength, Duration clientHelloTimeout) {
         throw new UnsupportedOperationException(
-                "sslConfig(ServerSslConfig, Map, int, Durations) is not supported by " + getClass());
+                "sslConfig(ServerSslConfig, Map, int, Duration) is not supported by " + getClass());
+    }
+
+    /**
+     * Set the SSL/TLS and <a href="https://tools.ietf.org/html/rfc6066#section-3">SNI</a> configuration.
+     *
+     * @param defaultConfig The configuration to use if the client certificate's SNI extension isn't present or the
+     * SNI hostname doesn't match any values in {@code sniMap}.
+     * @param sniMap A map where the keys are matched against the client certificate's SNI extension value in order
+     * to provide the corresponding {@link ServerSslConfig}.
+     * @param maxClientHelloLength The maximum length of a
+     * <a href="https://www.rfc-editor.org/rfc/rfc5246#section-7.4.1.2">ClientHello</a> message in bytes, up to
+     * {@code 2^24 - 1} bytes. Zero ({@code 0}) disables validation.
+     * @param clientHelloTimeout The timeout for waiting until
+     * <a href="https://www.rfc-editor.org/rfc/rfc5246#section-7.4.1.2">ClientHello</a> message is received.
+     * Implementations can round the specified {@link Duration} to full time units, depending on their time granularity.
+     * @param acceptInsecureConnections if non-TLS connections are accepted on the same socket.
+     * {@link Duration#ZERO Zero (0)} disables timeout.
+     * @return {@code this}.
+     */
+    default HttpServerBuilder sslConfig(ServerSslConfig defaultConfig, Map<String, ServerSslConfig> sniMap,
+                                        int maxClientHelloLength, Duration clientHelloTimeout,
+                                        boolean acceptInsecureConnections) {
```

Comment:
The new API lgtm! Let me just finish [discussion](https://github.com/apple/servicetalk/pull/2859#discussion_r1522546121) to make sure we don't miss anything.

> I'm going to go with the first approach, because with the second one it's again ambiguous (the default config does not have the acceptInsecureConnections).

Agreed, moving `acceptInsecureConnections` under `SniConfig` was likely confusing. WDYT about:

```java
HttpServerBuilder sslConfig(ServerSslConfig config);
HttpServerBuilder sslConfig(ServerSslConfig config, boolean acceptInsecureConnections);
HttpServerBuilder sslConfig(ServerSslConfig defaultConfig, SniConfig sniConfig);
HttpServerBuilder sslConfig(ServerSslConfig defaultConfig, boolean acceptInsecureConnections, SniConfig sniConfig);
@Deprecated
HttpServerBuilder sslConfig(ServerSslConfig defaultConfig, Map<String, ServerSslConfig> sniMap);
@Deprecated
HttpServerBuilder sslConfig(ServerSslConfig defaultConfig, Map<String, ServerSslConfig> sniMap, int maxClientHelloLength, Duration clientHelloTimeout);
```
My worry is that there could be more options for SNI in the future and it will force us to change API of `HttpServerBuilder` again and again. 

Another option to speed up this PR is to add only 
```java
HttpServerBuilder sslConfig(ServerSslConfig config, boolean acceptInsecureConnections);
```
And skip SNI for the follow-up, if we decide to go forward with `SniConfig` because it will require quite a bit of refactoring around existing SNI logic.

Reply:
@bryce-anderson we discussed offline with @daschl and agreed to add only 
```
HttpServerBuilder sslConfig(ServerSslConfig config, boolean acceptInsecureConnections);
``` 
in this PR. Could you please look at this thread from the [beginning](https://github.com/apple/servicetalk/pull/2859#discussion_r1522546121) and share your opinion? What will be a good API for SNI configuration?

Reply:
Correct, `TcpServerConfig` is public only due to packaging reasons. Unfortunately, we can not control how users use internal public API and therefore, must play a safe game for 0.42.x backward compatibility. No use-cases comes to my mind immediately except someone using a TCP server or building something on top of our TCP stack. 

> I would have thought `HttpServerBuilder sslConfig(ServerSslConfig defaultConfig, Map<String, ServerSslConfig> sniMap, boolean acceptInsecureConnections)` would have been the go-to setter for most people.


What about users who need 2 other options for SNI: `maxClientHelloLength` and `clientHelloTimeout`? Are you ok with yet another overload `HttpServerBuilder sslConfig(ServerSslConfig defaultConfig, Map<String, ServerSslConfig> sniMap, int maxClientHelloLength, Duration clientHelloTimeout, boolean acceptInsecureConnections)`? Or do you prefer to move `sniMap`, `maxClientHelloLength` and `clientHelloTimeout` under a new `SniConfig` with assumption that we will be able to add more props there if needed without changing `HttpServerBuilder` API again.